### DISCUSSION TITLE
What is your favourite config?

### DISCUSSION THREAD

**Discussion by vlad2305m** [vlad2305m/Sound-Physics-Fabric#2] [Show and tell]
What is your favourite config? and Can I make a preset without mentioning you?

**Comment by Luke322** [comment]
> What is your favourite config? and Can I make a preset without mentioning you?

The real question is wich config is used in the [video](https://youtu.be/qOQNSZBzD9Q)

**Reply by Rajbir98** [reply]
> > What is your favourite config? and Can I make a preset without mentioning you?
> 
> The real question is which config is used in the [video](https://youtu.be/qOQNSZBzD9Q)

I want to know that to

**Reply by ghost** [reply]
i want too

**Reply by FUDGEBUNGLER** [reply]
I believe that video was made with the default configuration when SonicEther initially released this mod back in 1.10/1.11 for Forge. Updating it to 1.17 Fabric seems to have broken the mod quite a bit.

**Reply by vlad2305m** [reply]
> I believe that video was made with the default configuration when SonicEther initially released this mod back in 1.10/1.11 for Forge. Updating it to 1.17 Fabric seems to have broken the mod quite a bit.

Well, I copied it as close as I could, and slapped some extra features on it, but you should consider that the libraries (and java version) changed (original had some issues with stereo) and the config is having issues with obfuscation )

**Comment by LEMMIIX** [comment]
Is there a config file I can upload or do I just post my settings? (screenshot or so)

**Reply by vlad2305m** [reply]
Thank you very much. A picture is certainly not sufficient. Please either rename to .txt or .zip and attach through drag-and-drop, or just copypaste your file in plain text, preferably as a collapsible object.
Have a nice day!

**Reply by LEMMIIX** [reply]
Alright. If you could help me locate my config file I'll be able to upload it here ^^

**Reply by vlad2305m** [reply]
Sure. Change resourcepack -> resourcepacks folder -> up to .minecraft, or whatever -> config -> sound-physics-config.json ?. Something like this.

**Reply by LEMMIIX** [reply]
Thanks, got it!
JSON as ZIP file attached

[LEMMIIX_SoundPhysics_config.zip](https://github.com/vlad2305m/Sound-Physics/files/6988373/LEMMIIX_SoundPhysics_config.zip)

Btw where do I report bugs/glitches?

**Reply by vlad2305m** [reply]
issues tab or anywhere I'll see them

**Comment by rght925h** [comment]
> 
> 
> What is your favourite config? and Can I make a preset without mentioning you?

I've spent some time to come up with a "natural" sounding config (lots of echo in caves, few to no echoes in grasslands, towns, dense forests, etc.). Likely still needs tweaking but this seems to work well so far:

[sound_physics_config.zip](https://github.com/vlad2305m/Sound-Physics-Fabric/files/7073209/sound_physics_config.zip)
_Adjust the performance options (64 rays, 5 bounces) to your system specs._

Try it out and use it / tweak it to your liking. Thank you for this incredibly great mod!

**Reply by HexHyte** [reply]
> [sound_physics_config.zip](https://github.com/vlad2305m/Sound-Physics-Fabric/files/7073209/sound_physics_config.zip)
> _Adjust the performance options (64 rays, 5 bounces) to your system specs._

Do you have an update for this? I've been working on a preset to share here but i am not still satisfied with the result

**Reply by rght925h** [reply]
> 
> 
> > [sound_physics_config.zip](https://github.com/vlad2305m/Sound-Physics-Fabric/files/7073209/sound_physics_config.zip)
> > _Adjust the performance options (64 rays, 5 bounces) to your system specs._
> 
> Do you have an update for this? I've been working on a preset to share here but i am not still satisfied with the result

I am still fine-tuning but the config as I uploaded it seems to work pretty well so far. Did you run into issues?

**Comment by thedocruby** [comment]
This is my favorite config, based on the one above. I spent the time to adjust every one of the reflectivity values. Also, I noticed that @rght925h had his config set so that you could hear mobs very clearly from very far away, so I also addressed this. The file is below:

[sound_physics_config.zip](https://github.com/vlad2305m/Sound-Physics-Fabric/files/7368750/sound_physics_config.zip)

While I was tweaking parameters and playtesting, I found that 64 rays was often not enough to consistently produce an accurate reverb effect. (you try to render a 360 panorama scene in blender using an 8 by 8 pixel canvas and tell me you can recognize any features) I increased to 128 rays and was met by a large performance drop. This seemed strange to me so I did some digging and found a bug in the code that rendered each of the 128 rays with as many bounces. This seemed to be the obvious source of the performance drop, and sure enough, setting the number of bounces to 1000 had zero effect on performance. So I fixed it, and some other things, and submitted a PR. After the bug is patched, my config should run without an issue, but it may not sound the same, because who knows how else that bug was affecting the mod. If necessary I will upload an updated version once the mod is patched.

Oh, and I would be honored for my config to be made into a preset, no credit necessary.

**Reply by vlad2305m** [reply]
Q: I changed MaxDirVariance to 1-MDV to avoid confusion. Did you want to redirect everything?

**Reply by rght925h** [reply]
Wonderful, thank you very much!

**Reply by thedocruby** [reply]
I updated it since the patch to include even more rays, because it seemed to still be helping. Apparently 256 rays still doesn't cause noticable performance issues, at least on my machine. Maybe it might during noisy situations, i.e. when raining. if necessary, you can turn it down again. Here is the file:
[sound_physics_config.zip](https://github.com/vlad2305m/Sound-Physics-Fabric/files/7403047/sound_physics_config.zip)

**Reply by FUDGEBUNGLER** [reply]
For some reason on my end the sound sources seem to be coming from randomly around the player? For example, footsteps don't sound centered in stereo and rain sounds like it is moving around the player when standing still. Is that just an issue with the mod or something that could be addressed in the config patch you released?

**Reply by vlad2305m** [reply]
> For some reason on my end the sound sources seem to be coming from randomly around the player? For example, footsteps don't sound centered in stereo and rain sounds like it is moving around the player when standing still. Is that just an issue with the mod or something that could be addressed in the config patch you released?

Yea, I reversed the behavior of direction variance for more intuitive use, but forgot to do so in default config. Just set it to 0.4 or turn it off.

**Reply by thedocruby** [reply]
> Yea, I reversed the behavior of direction variance for more intuitive use, but forgot to do so in default config. Just set it to 0.4 or turn it off.

I have experienced this too, what setting should i be changing to fix it?

**Reply by thedocruby** [reply]
@vlad2305m I figured it out, but don't forget to change that in the default config for the 1.18 release. I'll be using this mod with PlasmoVoice on a Youtuber SMP in the coming weeks. Thanks!

**Reply by vlad2305m** [reply]
Max dir variance. The formula is vector sum of inverse squares of distances (to the source, unfolded / total travelled) divided by the sum of inverse squares (aka mean). This gives length 1 if all directions are the same and 0 if they are opposite (I take square length for simplicity). That seems logical, but is confusing, so I did 1- to make it 0=same direction, 1=no dominant direction. If you are planning to do math, you will need this )

**Reply by thedocruby** [reply]
Thats actually pretty interesting, Thanks for sharing!!

**Comment by thedocruby** [comment]
**EDIT: The one i had here previously is just a renamed version of my old config, but it has now been fixed. (@GlowingUmbreon you may want to re-download it)**

@vlad2305m After playtesting my previous config for a couple months and tweaking it the entire time, I've retweaked it again and come up with this final version:
[DrRubisco_Signature.zip](https://github.com/vlad2305m/Sound-Physics-Fabric/files/7656240/DrRubisco_Signature.zip)

Please use the filename as the preset name in the config, thanks.
I also adjusted the max dir variance according to your changes, and it has helped a lot.

**Reply by GlowingUmbreon** [reply]
I've been playing with it a bit and I've notice there are no noticeable reverb, was this intentional or did I make a mistake?
Edit: I'm on 1.18 so maybe the update changed something which made it not work correctly?
Edit2: Acording to #16 the config file breaks the mod, also according to the change-log the default is your config (although I believe it is the older version).

**Reply by thedocruby** [reply]
@GlowingUmbreon yes, yes, and yes, I'll try and get to working on the reverb problem, not sure what thats about.

Edit: Apparently it is a 1.18 thing, I submitted an issue (#17 )
Double Edit: You were right, it is #16, it didn't seem like the same issue at first

**Comment by SoundEngineDevelopement** [comment]
> Well. Sonicether's mod was pathetic (mono audio, inefficient, etc), though I have no idea how to do reverb so I just copied it. I have already improved the logic a lot by almost completely rewriting it and I added some new features. I was happy with just any reverb, but DrRubisco tweaked some settings to make it very satisfying. He also significantly contributed to the logic. This mod is WIP, so there is a lot of features to come.

Mono audio really, no stereo? Didnt notice that when watching the showcase video on youtube. It sounded great and am sayin this as a soundcard user. Can i ask what you mean by reverb? Im not familiar with audio programming. Anyway im thankful for being able and use this soundengine as even when it's still WIP it does sound WAY better than vanilla audio.

Is there a version of this sound-engine available for minecraft forge? Fabric is not stable currently.

**Reply by GlowingUmbreon** [reply]
Please stop posting off topic stuff here, this discussion is for configs.

**Reply by SoundEngineDevelopement** [reply]
> Please stop posting off topic stuff here, this discussion is for configs.

Alright i will... where else can i ask then?
However is there a forge version of this mod is heavy mod releated question but yes not a config one...
Then again sorry if you beleve this is off topic.
This https://www.curseforge.com/minecraft/mc-mods/sound-physics-remastered is what i found for forge, would you confirm if it is releated to this mod or not? Cannot find anything else.

**Reply by vlad2305m** [reply]
It was forked somewhen earlier from this one. Seems to be maintained separately. Therefore features and bugs are different. Just start a new discussion or issue. And change your nickname.
